storage: plumb a tracing context into all queues' process()

BramGruneir opened the pull request:
Fixes #7535

<!-- Reviewable:start -->

---

This change is [<img src="https://reviewable.io/review_button.svg" height="35" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/7566)

<!-- Reviewable:end -->

BramGruneir committed:
storage: plumb a tracing context into all queues' process()

Fixes #7535

tamird commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 10 of 10 files at r1.
Review status: all files reviewed at latest revision, 9 unresolved discussions, all commit checks successful.

---

*[storage/gc_queue.go, line 294 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7566#-KLYN_n3BFjJK_NdQWpp:-KLYN_n4DEBrSIFEvxH_:-252073305) ([raw file](https://github.com/cockroachdb/cockroach/blob/8482eababac51f51759a79b05e30eaee68ecd54f/storage/gc_queue.go#L294)):*
> ```Go
> 	_ context.Context,
> ) error {
> 	ctx := repl.context(context.TODO())
> ```

use the context from above instead of this TODO?

---

*[storage/gc_queue_test.go, line 276 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7566#-KLYNf7wsXHAuHRDBo3r:-KLYNf7wsXHAuHRDBo3s:1350614369) ([raw file](https://github.com/cockroachdb/cockroach/blob/8482eababac51f51759a79b05e30eaee68ecd54f/storage/gc_queue_test.go#L276)):*
> ```Go
> 	// Process through a scan queue.
> 	gcQ := newGCQueue(tc.gossip)
> 	if err := gcQ.process(tc.clock.Now(), tc.rng, cfg, nil); err != nil {
> ```

seems not safe to pass nil.

---

*[storage/queue.go, line 114 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7566#-KLYNkdeGTKw-SQ_i7Z2:-KLYNkdeGTKw-SQ_i7Z3:-1708476619) ([raw file](https://github.com/cockroachdb/cockroach/blob/8482eababac51f51759a79b05e30eaee68ecd54f/storage/queue.go#L114)):*
> ```Go
> 
> 	// process accepts current time, a replica, and the system config
> 	// and executes queue-specific work on it. The context passed in is the
> ```

can you make the context the first argument? that's conventional.

I don't think you need to mention "which" context this is; there is only one, and it is essentially a bottomless bag of goodies.

---

*[storage/queue.go, line 437 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7566#-KLYP8q41cggbNtdrGjw:-KLYP8q41cggbNtdrGjx:-422579881) ([raw file](https://github.com/cockroachdb/cockroach/blob/8482eababac51f51759a79b05e30eaee68ecd54f/storage/queue.go#L437)):*
> ```Go
> 	}
> 
> 	var tracingCtx context.Context
> ```

just call it `ctx`

---

*[storage/queue.go, line 438 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7566#-KLYO3IpOq9CGl6ytAfM:-KLYO3IpOq9CGl6ytAfN:935136692) ([raw file](https://github.com/cockroachdb/cockroach/blob/8482eababac51f51759a79b05e30eaee68ecd54f/storage/queue.go#L438)):*
> ```Go
> 
> 	var tracingCtx context.Context
> 	if repl.store != nil {
> ```

whoa, how did this become necessary?

---

*[storage/raft_log_queue.go, line 132 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7566#-KLYP6nn_z5TAaLubX5w:-KLYP6nn_z5TAaLubX5x:1076766917) ([raw file](https://github.com/cockroachdb/cockroach/blob/8482eababac51f51759a79b05e30eaee68ecd54f/storage/raft_log_queue.go#L132)):*
> ```Go
> 	r *Replica,
> 	_ config.SystemConfig,
> 	tracingCtx context.Context,
> ```

just call it `ctx`. actually in this case it's `_`

---

*[storage/replica_gc_queue.go, line 139 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7566#-KLYPIb81uwkE4p4Uwfp:-KLYPIb81uwkE4p4Uwfq:-473638508) ([raw file](https://github.com/cockroachdb/cockroach/blob/8482eababac51f51759a79b05e30eaee68ecd54f/storage/replica_gc_queue.go#L139)):*
> ```Go
> 	rng *Replica,
> 	_ config.SystemConfig,
> 	tracingCtx context.Context,
> ```

`ctx`

---

*[storage/replicate_queue.go, line 112 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7566#-KLYPQVxOzLu4RL-zNQZ:-KLYPQVxOzLu4RL-zNQ_:160404226) ([raw file](https://github.com/cockroachdb/cockroach/blob/8482eababac51f51759a79b05e30eaee68ecd54f/storage/replicate_queue.go#L112)):*
> ```Go
> 	repl *Replica,
> 	sysCfg config.SystemConfig,
> 	tracingCtx context.Context,
> ```

ctx

---

*[storage/split_queue.go, line 94 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7566#-KLYPUaRLd0yHWwc4Wrs:-KLYPUaRLd0yHWwc4Wrt:160404226) ([raw file](https://github.com/cockroachdb/cockroach/blob/8482eababac51f51759a79b05e30eaee68ecd54f/storage/split_queue.go#L94)):*
> ```Go
> 	rng *Replica,
> 	sysCfg config.SystemConfig,
> 	tracingCtx context.Context,
> ```

ctx

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/7566#-:-KLYPau4m7b64cp9tD4G:1417088721)*
<!-- Sent from Reviewable.io -->

petermattis commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> modulo nits that `context.Context` is not tied to tracing and the style of trace messages.

---

Review status: all files reviewed at latest revision, 12 unresolved discussions, all commit checks successful.

---

*[storage/gc_queue_test.go, line 276 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7566#-KLYNf7wsXHAuHRDBo3r:-KLYucm5Ldg9rCOWQlZy:191822812) ([raw file](https://github.com/cockroachdb/cockroach/blob/8482eababac51f51759a79b05e30eaee68ecd54f/storage/gc_queue_test.go#L276)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> seems not safe to pass nil.
</details>
Ditto. Pass `context.TODO()`.

---

*[storage/queue.go, line 115 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7566#-KLYukKt4NF60tiPp51w:-KLYukKt4NF60tiPp51x:1473692706) ([raw file](https://github.com/cockroachdb/cockroach/blob/8482eababac51f51759a79b05e30eaee68ecd54f/storage/queue.go#L115)):*
> ```Go
> 	// process accepts current time, a replica, and the system config
> 	// and executes queue-specific work on it. The context passed in is the
> 	// tracing context.
> ```

The context is general and not specific to tracing.

---

*[storage/replicate_queue.go, line 133 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7566#-KLYvIBVst8gQWfG0_Ar:-KLYvIBW-R7qihUtUmTz:1568808726) ([raw file](https://github.com/cockroachdb/cockroach/blob/8482eababac51f51759a79b05e30eaee68ecd54f/storage/replicate_queue.go#L133)):*
> ```Go
> 	switch action {
> 	case AllocatorAdd:
> 		log.Trace(tracingCtx, "adding a new replica")
> ```

Include the range ID in the trace message.

---

*[storage/replicate_queue.go, line 144 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7566#-KLYvSC1VVfKnm9cbvJa:-KLYvSC2bsiI-tqZM6FV:-111182149) ([raw file](https://github.com/cockroachdb/cockroach/blob/8482eababac51f51759a79b05e30eaee68ecd54f/storage/replicate_queue.go#L144)):*
> ```Go
> 
> 		if log.V(1) {
> 			log.Infof("adding replica for under-replicated RangeID:%d to %+v", repl.RangeID, newReplica)
> ```

It would be good to standardize on message structure. Sometimes we say `range <id>`. Here you're using `RangeID:%d`. I prefer the former as it seems more consistent with other parts of the code base.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/7566#-:-KLYvwJhgdsTU-cQJy-x:526277377)*
<!-- Sent from Reviewable.io -->

tschottdorf commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> after everyone's nits.

---

Reviewed 10 of 10 files at r1.
Review status: all files reviewed at latest revision, 20 unresolved discussions, all commit checks successful.

---

*[storage/gc_queue.go, line 294 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7566#-KLYN_n3BFjJK_NdQWpp:-KLZqQEKYMH1uycKDlel:574457391) ([raw file](https://github.com/cockroachdb/cockroach/blob/8482eababac51f51759a79b05e30eaee68ecd54f/storage/gc_queue.go#L294)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> use the context from above instead of this TODO?
</details>
+1

Here and anywhere: as remarked on out-of-band preliminary review, the context goes first (this is a standard convention).

---

*[storage/queue.go, line 438 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7566#-KLYO3IpOq9CGl6ytAfM:-KLZr27H4P95z2VXyeUU:121299101) ([raw file](https://github.com/cockroachdb/cockroach/blob/8482eababac51f51759a79b05e30eaee68ecd54f/storage/queue.go#L438)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> whoa, how did this become necessary?
</details>
Probably for some tests, in which case I'd prefer adapting the tests or at least adding commentary here.

---

*[storage/replica_gc_queue.go, line 182 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7566#-KLZrJamWRUNqj4izdeh:-KLZrJamWRUNqj4izdei:30047116) ([raw file](https://github.com/cockroachdb/cockroach/blob/8482eababac51f51759a79b05e30eaee68ecd54f/storage/replica_gc_queue.go#L182)):*
> ```Go
> 			log.Infof("destroying local data from range %d", desc.RangeID)
> 		}
> 		log.Trace(tracingCtx, fmt.Sprintf("destroying local data from range %d", desc.RangeID))
> ```

as discussed, just log the rangeID once right after you create the trace (in baseQueue) and then you can use fixed strings for most purposes here.

---

*[storage/replicate_queue.go, line 133 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7566#-KLYvIBVst8gQWfG0_Ar:-KLZrU9_1HGyad9seRtr:409955980) ([raw file](https://github.com/cockroachdb/cockroach/blob/8482eababac51f51759a79b05e30eaee68ecd54f/storage/replicate_queue.go#L133)):*
<details><summary><i>Previously, petermattis (Peter Mattis) wrote…</i></summary>
> Include the range ID in the trace message.
</details>
The rangeID should be logged in baseQueue, where the context is populated. That way, we don't need to redundantly do this for all queue impls.

---

*[storage/replicate_queue.go, line 144 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7566#-KLYvSC1VVfKnm9cbvJa:-KLZrbtBjXCP1qRWLywo:1388437497) ([raw file](https://github.com/cockroachdb/cockroach/blob/8482eababac51f51759a79b05e30eaee68ecd54f/storage/replicate_queue.go#L144)):*
<details><summary><i>Previously, petermattis (Peter Mattis) wrote…</i></summary>
> It would be good to standardize on message structure. Sometimes we say `range <id>`. Here you're using `RangeID:%d`. I prefer the former as it seems more consistent with other parts of the code base.
</details>
RangeID should go away here (log only once in baseQueue). I also prefer `range <id>` as that's de facto standard atm.

---

*[storage/replicate_queue.go, line 146 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7566#-KLZrjXGg_gYQUBVx1Zf:-KLZrjXHjzNWbV6ez9wF:1904222067) ([raw file](https://github.com/cockroachdb/cockroach/blob/8482eababac51f51759a79b05e30eaee68ecd54f/storage/replicate_queue.go#L146)):*
> ```Go
> 			log.Infof("adding replica for under-replicated RangeID:%d to %+v", repl.RangeID, newReplica)
> 		}
> 		log.Trace(tracingCtx, fmt.Sprintf("adding replica for under-replicated RangeID:%d to %+v", repl.RangeID, newReplica))
> ```

ditto no RangeID (here and everywhere, probably left out a few).

---

*[storage/replicate_queue.go, line 159 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7566#-KLZrnQB6L5mvZl0kLdy:-KLZrnQB6L5mvZl0kLdz:-2046481826) ([raw file](https://github.com/cockroachdb/cockroach/blob/8482eababac51f51759a79b05e30eaee68ecd54f/storage/replicate_queue.go#L159)):*
> ```Go
> 			log.Infof("removing replica for over-replicated RangeID:%d from %+v", repl.RangeID, removeReplica)
> 		}
> 		log.Trace(tracingCtx, fmt.Sprintf("removing replica for over-replicated RangeID:%d from %+v", repl.RangeID, removeReplica))
> ```

no RangeID

---

*[storage/replicate_queue.go, line 178 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7566#-KLZrqCk8iTTnm9ERjEG:-KLZrqCk8iTTnm9ERjEH:-469765787) ([raw file](https://github.com/cockroachdb/cockroach/blob/8482eababac51f51759a79b05e30eaee68ecd54f/storage/replicate_queue.go#L178)):*
> ```Go
> 			log.Infof("removing replica from dead store RangeID:%d from %+v", repl.RangeID, deadReplicas[0])
> 		}
> 		log.Trace(tracingCtx, fmt.Sprintf("removing replica from dead store RangeID:%d from %+v", repl.RangeID, deadReplicas[0]))
> ```

ditto

---

*[storage/replicate_queue.go, line 203 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7566#-KLZrt4ezUJ85pDfkTVd:-KLZrt4ezUJ85pDfkTVe:-469765787) ([raw file](https://github.com/cockroachdb/cockroach/blob/8482eababac51f51759a79b05e30eaee68ecd54f/storage/replicate_queue.go#L203)):*
> ```Go
> 			log.Infof("rebalancing RangeID:%d to %+v", repl.RangeID, rebalanceReplica)
> 		}
> 		log.Trace(tracingCtx, fmt.Sprintf("rebalancing RangeID:%d to %+v", repl.RangeID, rebalanceReplica))
> ```

ditto

---

*[storage/replicate_queue.go, line 209 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7566#-KLZrtu8n4MTizUSIVu3:-KLZrtu8n4MTizUSIVu4:63600010) ([raw file](https://github.com/cockroachdb/cockroach/blob/8482eababac51f51759a79b05e30eaee68ecd54f/storage/replicate_queue.go#L209)):*
> ```Go
> 	}
> 
> 	log.Trace(tracingCtx, "operation complete")
> ```

this should be in the base queue so that it's the same for all ops.

---

*[storage/split_queue.go, line 103 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7566#-KLZs2qbSd92KUoX68Pc:-KLZs2qbSd92KUoX68Pd:-768462294) ([raw file](https://github.com/cockroachdb/cockroach/blob/8482eababac51f51759a79b05e30eaee68ecd54f/storage/split_queue.go#L103)):*
> ```Go
> 	if len(splitKeys) > 0 {
> 		log.Infof("splitting %s at keys %v", rng, splitKeys)
> 		log.Trace(tracingCtx, fmt.Sprintf("splitting %s at keys %v", rng, splitKeys))
> ```

`rng` as `%s` can simply be logged at `baseQueue` (as a more elaborate version of the RangeID I've been suggesting throughout)

---

*[storage/split_queue.go, line 121 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7566#-KLZsB9MiT88mZ0Q5NyA:-KLZsB9MiT88mZ0Q5NyB:-469765787) ([raw file](https://github.com/cockroachdb/cockroach/blob/8482eababac51f51759a79b05e30eaee68ecd54f/storage/split_queue.go#L121)):*
> ```Go
> 	if float64(size)/float64(zone.RangeMaxBytes) > 1 {
> 		log.Infof("splitting %s size=%d max=%d", rng, size, zone.RangeMaxBytes)
> 		log.Trace(tracingCtx, fmt.Sprintf("splitting %s size=%d max=%d", rng, size, zone.RangeMaxBytes))
> ```

ditto

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/7566#-:-KLZsEzxXahrEwnzXmZ-:-1659056034)*
<!-- Sent from Reviewable.io -->

BramGruneir commented:
Review status: 0 of 10 files reviewed at latest revision, 20 unresolved discussions, some commit checks pending.

---

*[storage/gc_queue.go, line 294 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7566#-KLYN_n3BFjJK_NdQWpp:-KLc8_5K-J4rLKJ4iKvD:677588487) ([raw file](https://github.com/cockroachdb/cockroach/blob/8482eababac51f51759a79b05e30eaee68ecd54f/storage/gc_queue.go#L294)):*
<details><summary><i>Previously, tschottdorf (Tobias Schottdorf) wrote…</i></summary>
> +1
> 
> Here and anywhere: as remarked on out-of-band preliminary review, the context goes first (this is a standard convention).
</details>
done

---

*[storage/gc_queue_test.go, line 276 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7566#-KLYNf7wsXHAuHRDBo3r:-KLcMpKL0_OeMqEU20Uw:1880210541) ([raw file](https://github.com/cockroachdb/cockroach/blob/8482eababac51f51759a79b05e30eaee68ecd54f/storage/gc_queue_test.go#L276)):*
<details><summary><i>Previously, petermattis (Peter Mattis) wrote…</i></summary>
> Ditto. Pass `context.TODO()`.
</details>
Done.  I just went back through the documentation on context, never use nil for a context is the general rule.

---

*[storage/queue.go, line 114 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7566#-KLYNkdeGTKw-SQ_i7Z2:-KLcMs1_8B1vYyoxCTvz:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/8482eababac51f51759a79b05e30eaee68ecd54f/storage/queue.go#L114)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> can you make the context the first argument? that's conventional.
> 
> I don't think you need to mention "which" context this is; there is only one, and it is essentially a bottomless bag of goodies.
</details>
Done.

---

*[storage/queue.go, line 115 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7566#-KLYukKt4NF60tiPp51w:-KLcMvAsVZOLj2litCob:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/8482eababac51f51759a79b05e30eaee68ecd54f/storage/queue.go#L115)):*
<details><summary><i>Previously, petermattis (Peter Mattis) wrote…</i></summary>
> The context is general and not specific to tracing.
</details>
Done.

---

*[storage/queue.go, line 437 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7566#-KLYP8q41cggbNtdrGjw:-KLvhxjrKjcpYjhCN9Yh:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/8482eababac51f51759a79b05e30eaee68ecd54f/storage/queue.go#L437)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> just call it `ctx`
</details>
Done.

---

*[storage/queue.go, line 438 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7566#-KLYO3IpOq9CGl6ytAfM:-KLvgNzCklRfq1tjChZf:82310542) ([raw file](https://github.com/cockroachdb/cockroach/blob/8482eababac51f51759a79b05e30eaee68ecd54f/storage/queue.go#L438)):*
<details><summary><i>Previously, tschottdorf (Tobias Schottdorf) wrote…</i></summary>
> Probably for some tests, in which case I'd prefer adapting the tests or at least adding commentary here.
</details>
Most of the queue tests didn't have a store attached to them.  I started by plumbing the tracer directly into the queue, but that was just too messy.  I've adapted the tests instead of adding this check.

---

*[storage/raft_log_queue.go, line 132 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7566#-KLYP6nn_z5TAaLubX5w:-KLvi0w3BhjVm7U0ZCJh:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/8482eababac51f51759a79b05e30eaee68ecd54f/storage/raft_log_queue.go#L132)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> just call it `ctx`. actually in this case it's `_`
</details>
Done.

---

*[storage/replica_gc_queue.go, line 139 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7566#-KLYPIb81uwkE4p4Uwfp:-KLvi5HahlFtcJApTDVI:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/8482eababac51f51759a79b05e30eaee68ecd54f/storage/replica_gc_queue.go#L139)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> `ctx`
</details>
Done.

---

*[storage/replica_gc_queue.go, line 182 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7566#-KLZrJamWRUNqj4izdeh:-KLviGQbJpEWxz7YXje7:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/8482eababac51f51759a79b05e30eaee68ecd54f/storage/replica_gc_queue.go#L182)):*
<details><summary><i>Previously, tschottdorf (Tobias Schottdorf) wrote…</i></summary>
> as discussed, just log the rangeID once right after you create the trace (in baseQueue) and then you can use fixed strings for most purposes here.
</details>
Done.

---

*[storage/replicate_queue.go, line 112 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7566#-KLYPQVxOzLu4RL-zNQZ:-KLvkPWNX6X6ABUM7JHO:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/8482eababac51f51759a79b05e30eaee68ecd54f/storage/replicate_queue.go#L112)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> ctx
</details>
Done.

---

*[storage/replicate_queue.go, line 133 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7566#-KLYvIBVst8gQWfG0_Ar:-KLvis_IKEq8JpwLBrgr:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/8482eababac51f51759a79b05e30eaee68ecd54f/storage/replicate_queue.go#L133)):*
<details><summary><i>Previously, tschottdorf (Tobias Schottdorf) wrote…</i></summary>
> The rangeID should be logged in baseQueue, where the context is populated. That way, we don't need to redundantly do this for all queue impls.
</details>
Done.

---

*[storage/replicate_queue.go, line 144 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7566#-KLYvSC1VVfKnm9cbvJa:-KLvivHN1ViLK7q0lj9k:1948068022) ([raw file](https://github.com/cockroachdb/cockroach/blob/8482eababac51f51759a79b05e30eaee68ecd54f/storage/replicate_queue.go#L144)):*
<details><summary><i>Previously, tschottdorf (Tobias Schottdorf) wrote…</i></summary>
> RangeID should go away here (log only once in baseQueue). I also prefer `range <id>` as that's de facto standard atm.
</details>
The reason I prefer RangeID:%d is because it matches what %+v outputs when you output a full descriptor.  But I've changed all these messages to range <id>.

---

*[storage/replicate_queue.go, line 146 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7566#-KLZrjXGg_gYQUBVx1Zf:-KLvjv7KLXssD82A49En:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/8482eababac51f51759a79b05e30eaee68ecd54f/storage/replicate_queue.go#L146)):*
<details><summary><i>Previously, tschottdorf (Tobias Schottdorf) wrote…</i></summary>
> ditto no RangeID (here and everywhere, probably left out a few).
</details>
Done.

---

*[storage/replicate_queue.go, line 159 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7566#-KLZrnQB6L5mvZl0kLdy:-KLvk-WG3YPIWK2VS7yY:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/8482eababac51f51759a79b05e30eaee68ecd54f/storage/replicate_queue.go#L159)):*
<details><summary><i>Previously, tschottdorf (Tobias Schottdorf) wrote…</i></summary>
> no RangeID
</details>
Done.

---

*[storage/replicate_queue.go, line 178 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7566#-KLZrqCk8iTTnm9ERjEG:-KLvk08WmqSfwfnbfEdM:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/8482eababac51f51759a79b05e30eaee68ecd54f/storage/replicate_queue.go#L178)):*
<details><summary><i>Previously, tschottdorf (Tobias Schottdorf) wrote…</i></summary>
> ditto
</details>
Done.

---

*[storage/replicate_queue.go, line 203 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7566#-KLZrt4ezUJ85pDfkTVd:-KLvkSM4bnubMIr2MVIp:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/8482eababac51f51759a79b05e30eaee68ecd54f/storage/replicate_queue.go#L203)):*
<details><summary><i>Previously, tschottdorf (Tobias Schottdorf) wrote…</i></summary>
> ditto
</details>
Done.

---

*[storage/replicate_queue.go, line 209 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7566#-KLZrtu8n4MTizUSIVu3:-KLvk8Mi7vPopxYSisBG:-1937638425) ([raw file](https://github.com/cockroachdb/cockroach/blob/8482eababac51f51759a79b05e30eaee68ecd54f/storage/replicate_queue.go#L209)):*
<details><summary><i>Previously, tschottdorf (Tobias Schottdorf) wrote…</i></summary>
> this should be in the base queue so that it's the same for all ops.
</details>
just removed it, no need

---

*[storage/split_queue.go, line 94 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7566#-KLYPUaRLd0yHWwc4Wrs:-KLvkIO0XplvmncyLoLF:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/8482eababac51f51759a79b05e30eaee68ecd54f/storage/split_queue.go#L94)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> ctx
</details>
Done.

---

*[storage/split_queue.go, line 103 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7566#-KLZs2qbSd92KUoX68Pc:-KLvkKFUTJvCd6hgsNHe:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/8482eababac51f51759a79b05e30eaee68ecd54f/storage/split_queue.go#L103)):*
<details><summary><i>Previously, tschottdorf (Tobias Schottdorf) wrote…</i></summary>
> `rng` as `%s` can simply be logged at `baseQueue` (as a more elaborate version of the RangeID I've been suggesting throughout)
</details>
Done.

---

*[storage/split_queue.go, line 121 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7566#-KLZsB9MiT88mZ0Q5NyA:-KLvkLRCXqsQLYw7BSxQ:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/8482eababac51f51759a79b05e30eaee68ecd54f/storage/split_queue.go#L121)):*
<details><summary><i>Previously, tschottdorf (Tobias Schottdorf) wrote…</i></summary>
> ditto
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/7566)*
<!-- Sent from Reviewable.io -->

tamird commented:
Reviewed 10 of 10 files at r2.
Review status: all files reviewed at latest revision, 16 unresolved discussions, some commit checks failed.

---

*[storage/queue.go, line 438 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7566#-KLw0VVEhjKs_r3FIOq6:-KLw0VVEhjKs_r3FIOq7:-1300181522) ([raw file](https://github.com/cockroachdb/cockroach/blob/cafac4862fa64142cc562edde425275f885757a7/storage/queue.go#L438)):*
> ```Go
> 	sp := repl.store.Tracer().StartSpan(bq.name)
> 	ctx := opentracing.ContextWithSpan(repl.context(context.Background()), sp)
> 	log.Trace(ctx, fmt.Sprintf("queue start for range %d", repl.RangeID))
> ```

do you need the Sprintf here? I thought that `repl.context()` already attached this information to the context.

---

*[storage/queue_test.go, line 273 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7566#-KLw15vGj0M9Mw-Fp0F5:-KLw15vGj0M9Mw-Fp0F6:-956053783) ([raw file](https://github.com/cockroachdb/cockroach/blob/cafac4862fa64142cc562edde425275f885757a7/storage/queue_test.go#L273)):*
> ```Go
> 		t.Fatal(err)
> 	}
> 	r1.store = tc.store
> ```

throughout: why don't you put this up in the literal?

---

*[storage/replicate_queue.go, line 146 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7566#-KLZrjXGg_gYQUBVx1Zf:-KLw1YjhlUWcQ-ZsSNXR:-593034279) ([raw file](https://github.com/cockroachdb/cockroach/blob/8482eababac51f51759a79b05e30eaee68ecd54f/storage/replicate_queue.go#L146)):*
<details><summary><i>Previously, BramGruneir (Bram Gruneir) wrote…</i></summary>
> Done.
</details>
throughout this file: there's hardly anything between this trace line and the preceding one; is that what you intended to do?
throughout this file: please double check the grammar of all the trace lines you added in this file.

everywhere: are these Sprintf calls needed? if so, you should fmt.Sprintf just once for both the trace and the log, to avoid needless allocations.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/7566)*
<!-- Sent from Reviewable.io -->

BramGruneir commented:
Review status: all files reviewed at latest revision, 16 unresolved discussions, some commit checks failed.

---

*[storage/queue.go, line 438 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7566#-KLw0VVEhjKs_r3FIOq6:-KLwTgsps_vi7wujTNv9:484081166) ([raw file](https://github.com/cockroachdb/cockroach/blob/cafac4862fa64142cc562edde425275f885757a7/storage/queue.go#L438)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> do you need the Sprintf here? I thought that `repl.context()` already attached this information to the context.
</details>
Without this, there's not info in the trace about which range was being rebalanced.  The other option is to name the span to include the range id.

---

*[storage/queue_test.go, line 273 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7566#-KLw15vGj0M9Mw-Fp0F5:-KLwYOyh4huBpFiXZq-j:-789792320) ([raw file](https://github.com/cockroachdb/cockroach/blob/cafac4862fa64142cc562edde425275f885757a7/storage/queue_test.go#L273)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> throughout: why don't you put this up in the literal?
</details>
It was due to calling setDesc which tried to process the replica (which is uninitialized).  But the setDescWithoutProcessUpdate works perfectly.  Cleaned up all tests.

---

*[storage/replicate_queue.go, line 146 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7566#-KLZrjXGg_gYQUBVx1Zf:-KLwZbMY7-SMyRrHkdyf:517848604) ([raw file](https://github.com/cockroachdb/cockroach/blob/8482eababac51f51759a79b05e30eaee68ecd54f/storage/replicate_queue.go#L146)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> throughout this file: there's hardly anything between this trace line and the preceding one; is that what you intended to do?
> throughout this file: please double check the grammar of all the trace lines you added in this file.
> 
> everywhere: are these Sprintf calls needed? if so, you should fmt.Sprintf just once for both the trace and the log, to avoid needless allocations.
</details>
1) Yes, this is intended, to see how long the allocator takes
2) The logs include the range id while the trace does not, so there's no optimal way to do this.
3) done

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/7566)*
<!-- Sent from Reviewable.io -->

tamird commented:
Reviewed 3 of 3 files at r3.
Review status: all files reviewed at latest revision, 17 unresolved discussions, all commit checks successful.

---

*[storage/queue.go, line 438 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7566#-KLw0VVEhjKs_r3FIOq6:-KLwg96eCPLhQPD6eLZv:345302035) ([raw file](https://github.com/cockroachdb/cockroach/blob/cafac4862fa64142cc562edde425275f885757a7/storage/queue.go#L438)):*
<details><summary><i>Previously, BramGruneir (Bram Gruneir) wrote…</i></summary>
> Without this, there's not info in the trace about which range was being rebalanced.  The other option is to name the span to include the range id.
</details>
OK, so `repl.Context()` doesn't do what i thought :(

---

*[storage/queue.go, line 453 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7566#-KLwgj57RIwkhTpSNPkZ:-KLwgj58k9N-w_G-7Fvt:1484413609) ([raw file](https://github.com/cockroachdb/cockroach/blob/e215be32136e7565173c34ff3a3b774ec8993e0e/storage/queue.go#L453)):*
> ```Go
> 			return errors.Wrapf(err.GoError(), "%s: could not obtain lease", repl)
> 		}
> 		log.Trace(ctx, "got range lease")
> ```

given that you say "queue start for ..." above, does this similarly need such context? ditto for the "done" below.

---

*[storage/queue_test.go, line 74 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7566#-KLwhBiqY6SMWXXugHBT:-KLwhBiqY6SMWXXugHBU:1530623742) ([raw file](https://github.com/cockroachdb/cockroach/blob/e215be32136e7565173c34ff3a3b774ec8993e0e/storage/queue_test.go#L74)):*
> ```Go
> 	pChan         chan struct{}
> 	mu            struct {
> 		sync.Mutex
> ```

what made this necessary?

---

*[storage/replicate_queue.go, line 146 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7566#-KLZrjXGg_gYQUBVx1Zf:-KLwhSj0weRG8XUjZU6Q:1110106200) ([raw file](https://github.com/cockroachdb/cockroach/blob/8482eababac51f51759a79b05e30eaee68ecd54f/storage/replicate_queue.go#L146)):*
<details><summary><i>Previously, BramGruneir (Bram Gruneir) wrote…</i></summary>
> 1) Yes, this is intended, to see how long the allocator takes
> 2) The logs include the range id while the trace does not, so there's no optimal way to do this.
> 3) done
</details>
...are your replies out of order?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/7566)*
<!-- Sent from Reviewable.io -->

BramGruneir commented:
Review status: all files reviewed at latest revision, 16 unresolved discussions, all commit checks successful.

---

*[storage/queue.go, line 453 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7566#-KLwgj57RIwkhTpSNPkZ:-KLwj3DRBLlWAQgOXEPK:-1537984784) ([raw file](https://github.com/cockroachdb/cockroach/blob/e215be32136e7565173c34ff3a3b774ec8993e0e/storage/queue.go#L453)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> given that you say "queue start for ..." above, does this similarly need such context? ditto for the "done" below.
</details>
yes, these are tracking how long it takes to get the range lease and how long the processing takes as a whole.  It makes looking at the tracing output much easier to digest.

---

*[storage/queue_test.go, line 74 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7566#-KLwhBiqY6SMWXXugHBT:-KLwnEvcAoUPrkIXf32G:1371022331) ([raw file](https://github.com/cockroachdb/cockroach/blob/e215be32136e7565173c34ff3a3b774ec8993e0e/storage/queue_test.go#L74)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> what made this necessary?
</details>
There was a race on err.  I put processed in there as well to make the implementation cleaner.

---

*[storage/replicate_queue.go, line 146 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7566#-KLZrjXGg_gYQUBVx1Zf:-KLwnQcUJLMRFNYSC9xy:594336686) ([raw file](https://github.com/cockroachdb/cockroach/blob/8482eababac51f51759a79b05e30eaee68ecd54f/storage/replicate_queue.go#L146)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> ...are your replies out of order?
</details>
Yes.  1,3,2 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/7566)*
<!-- Sent from Reviewable.io -->

BramGruneir commented:
Review status: 7 of 10 files reviewed at latest revision, 16 unresolved discussions, some commit checks pending.

---

*[storage/queue_test.go, line 74 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7566#-KLwhBiqY6SMWXXugHBT:-KM-z7qWUrLm_nxthGeL:-186746903) ([raw file](https://github.com/cockroachdb/cockroach/blob/e215be32136e7565173c34ff3a3b774ec8993e0e/storage/queue_test.go#L74)):*
<details><summary><i>Previously, BramGruneir (Bram Gruneir) wrote…</i></summary>
> There was a race on err.  I put processed in there as well to make the implementation cleaner.
</details>
The race shouldn't happen anymore now that #7645 has been merged.  So I reverted the sync changes.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/7566)*
<!-- Sent from Reviewable.io -->

tamird commented:
Reviewed 1 of 1 files at r4, 2 of 2 files at r5.
Review status: all files reviewed at latest revision, 19 unresolved discussions, some commit checks pending.

---

*[storage/gc_queue_test.go, line 497 \[r5\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7566#-KM0--0xIj38vsBsL8ut:-KM0--0xIj38vsBsL8uu:-300521986) ([raw file](https://github.com/cockroachdb/cockroach/blob/4840288d17c2cfb670dba7a39ec71f57287eb71f/storage/gc_queue_test.go#L497)):*
> ```Go
> 	}
> 
> 	if err := gcQ.process(context.TODO(), tc.clock.Now(), tc.rng, cfg); err != nil {
> ```

throughout tests: use `context.Background()` - nobody is going to fix this "TODO"

---

*[storage/queue.go, line 453 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7566#-KLwgj57RIwkhTpSNPkZ:-KLwq5ZNppxCISdnBFlz:1563212718) ([raw file](https://github.com/cockroachdb/cockroach/blob/e215be32136e7565173c34ff3a3b774ec8993e0e/storage/queue.go#L453)):*
<details><summary><i>Previously, BramGruneir (Bram Gruneir) wrote…</i></summary>
> yes, these are tracking how long it takes to get the range lease and how long the processing takes as a whole.  It makes looking at the tracing output much easier to digest.
</details>
huh? yes they need context? i'm confused.

---

*[storage/queue_test.go, line 500 \[r5\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7566#-KM0-IVRd_rVjnYZsw5W:-KM0-IVRd_rVjnYZsw5X:901436166) ([raw file](https://github.com/cockroachdb/cockroach/blob/4840288d17c2cfb670dba7a39ec71f57287eb71f/storage/queue_test.go#L500)):*
> ```Go
> 		// placed in purgatory.
> 		// Verify that the size of the purgatory map is correct.
> 		if l := bq.PurgatoryLength(); l != replicaCount {
> ```

this is no longer atomic with the priority queue length below. is that expected/okay?

---

*[storage/queue_test.go, line 521 \[r5\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7566#-KM0-MKBrIeJu_bVGp_e:-KM0-MKBrIeJu_bVGp_f:901436166) ([raw file](https://github.com/cockroachdb/cockroach/blob/4840288d17c2cfb670dba7a39ec71f57287eb71f/storage/queue_test.go#L521)):*
> ```Go
> 		// placed in purgatory.
> 		// Verify the replicas are still in purgatory.
> 		if l := bq.PurgatoryLength(); l != replicaCount {
> ```

this is no longer atomic with the priority queue length below. is that expected/okay?

---

*[storage/queue_test.go, line 543 \[r5\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7566#-KM0-SEAVN5Mqm6IWBL4:-KM0-SEAVN5Mqm6IWBL5:901436166) ([raw file](https://github.com/cockroachdb/cockroach/blob/4840288d17c2cfb670dba7a39ec71f57287eb71f/storage/queue_test.go#L543)):*
> ```Go
> 
> 	// Verify the replicas are no longer in purgatory.
> 	if l := bq.PurgatoryLength(); l != 0 {
> ```

this is no longer atomic with the priority queue length below. is that expected/okay?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/7566)*
<!-- Sent from Reviewable.io -->

BramGruneir commented:
Review status: all files reviewed at latest revision, 19 unresolved discussions, all commit checks successful.

---

*[storage/gc_queue_test.go, line 497 \[r5\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7566#-KM0--0xIj38vsBsL8ut:-KM02pldcek5S3jtR1KV:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/4840288d17c2cfb670dba7a39ec71f57287eb71f/storage/gc_queue_test.go#L497)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> throughout tests: use `context.Background()` - nobody is going to fix this "TODO"
</details>
Done.

---

*[storage/queue.go, line 453 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7566#-KLwgj57RIwkhTpSNPkZ:-KM02wEh6pMcXnCt5be3:-412341374) ([raw file](https://github.com/cockroachdb/cockroach/blob/e215be32136e7565173c34ff3a3b774ec8993e0e/storage/queue.go#L453)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> huh? yes they need context? i'm confused.
</details>
Sorry, I think I was replying to something else. 
No, they don't need context.  It's clear from tracing output.

---

*[storage/queue_test.go, line 500 \[r5\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7566#-KM0-IVRd_rVjnYZsw5W:-KM02k3V5SjRSKNlIScx:1293779113) ([raw file](https://github.com/cockroachdb/cockroach/blob/4840288d17c2cfb670dba7a39ec71f57287eb71f/storage/queue_test.go#L500)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> this is no longer atomic with the priority queue length below. is that expected/okay?
</details>
Perfectly fine.

---

*[storage/queue_test.go, line 521 \[r5\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7566#-KM0-MKBrIeJu_bVGp_e:-KM02mGAHKZW_it5IOvx:1293779113) ([raw file](https://github.com/cockroachdb/cockroach/blob/4840288d17c2cfb670dba7a39ec71f57287eb71f/storage/queue_test.go#L521)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> this is no longer atomic with the priority queue length below. is that expected/okay?
</details>
Perfectly fine.

---

*[storage/queue_test.go, line 543 \[r5\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7566#-KM0-SEAVN5Mqm6IWBL4:-KM02oIbyTwOwODAWJXo:1293779113) ([raw file](https://github.com/cockroachdb/cockroach/blob/4840288d17c2cfb670dba7a39ec71f57287eb71f/storage/queue_test.go#L543)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> this is no longer atomic with the priority queue length below. is that expected/okay?
</details>
Perfectly fine.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/7566)*
<!-- Sent from Reviewable.io -->

tamird commented:
Review status: all files reviewed at latest revision, 15 unresolved discussions, all commit checks successful.

---

*[storage/gc_queue_test.go, line 497 \[r5\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7566#-KM0--0xIj38vsBsL8ut:-KM0GPQLnXdA3jbSZGP8:567050842) ([raw file](https://github.com/cockroachdb/cockroach/blob/4840288d17c2cfb670dba7a39ec71f57287eb71f/storage/gc_queue_test.go#L497)):*
<details><summary><i>Previously, BramGruneir (Bram Gruneir) wrote…</i></summary>
> Done.
</details>
doesn't look done. lgtm otherwise

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/7566)*
<!-- Sent from Reviewable.io -->

BramGruneir commented:
Review status: 9 of 10 files reviewed at latest revision, 15 unresolved discussions, some commit checks pending.

---

*[storage/gc_queue_test.go, line 497 \[r5\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7566#-KM0--0xIj38vsBsL8ut:-KM0REx4qGfR9UYnnS9F:-656769128) ([raw file](https://github.com/cockroachdb/cockroach/blob/4840288d17c2cfb670dba7a39ec71f57287eb71f/storage/gc_queue_test.go#L497)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> doesn't look done. lgtm otherwise
</details>
Forgot to push.  Sorry.  

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/7566)*
<!-- Sent from Reviewable.io -->

tamird commented:
Reviewed 1 of 1 files at r6.
Review status: all files reviewed at latest revision, 14 unresolved discussions, some commit checks pending.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/7566)*
<!-- Sent from Reviewable.io -->
